pandas/_typing.py

```diff
@@ -23,21 +23,29 @@
     from pandas.core.indexes.base import Index  # noqa: F401
     from pandas.core.series import Series  # noqa: F401
     from pandas.core.generic import NDFrame  # noqa: F401
+    from pandas import Interval  # noqa: F401
 
+# array-like
 
 AnyArrayLike = TypeVar("AnyArrayLike", "ExtensionArray", "Index", "Series", np.ndarray)
 ArrayLike = TypeVar("ArrayLike", "ExtensionArray", np.ndarray)
+
+# scalars
+
+PythonScalar = Union[str, int, float, bool]
 DatetimeLikeScalar = TypeVar("DatetimeLikeScalar", "Period", "Timestamp", "Timedelta")
+PandasScalar = Union[DatetimeLikeScalar, "Interval"]
+Scalar = Union[PythonScalar, PandasScalar]
```

Comment:
Does it matter that this is a Union of a Union and a TypeVar? Maybe DatetimeLikeScalar should just be a Union?

Reply:
yes, we have the same issue as with FilePathOrBuffer, where IO is unbound in the Union and requires the addition of type parameters when using the alias.

and we don't really want that for Scalar. but we had Scalar as a union (for JSONSerializable)  and DatetimeLikeScalar as a TypeVar.

> Maybe DatetimeLikeScalar should just be a Union?

rather than change this, maybe should define `PandasScalar = Union["Period", "Timestamp", "Timedelta", "Interval"]` for now

Reply:
Yea makes sense